src/mz390.java

```diff
@@ -3210,7 +3220,44 @@ private boolean zsm_find_name(){
 			return false;
 		}
 	}
-	private void parse_mac_line(){
+	private void insert_acall_parms(){  // #335
+		/*
+		/* insert &(acall_name)(parm#) SETC 'parm' for each parm in acall name(p1,p2,,pn)	
+		 */
+		 if (acall_parm.length() < 3)return; // #335 last fix for v1.8.1
+		 String acall_arg = acall_parm.substring(1,acall_parm.length()-1);
+		 String acall_arg_insert = "";
+		 int acall_arg_comma = 0;
+		 int acall_arg_num = 1;
+		 acall_arg_comma = acall_arg.indexOf(',');
+		 while (acall_arg_comma > 0){
+			   if (!acall_arg.substring(0,1).equals("\'")){  // #335
+		          acall_arg_insert = "&(" + acall_name + ")(" + acall_arg_num + ") SETC \'" + acall_arg.substring(0,acall_arg_comma) + "\'";
+		       } else {
+				  acall_arg_insert = "&(" + acall_name + ")(" + acall_arg_num + ") SETC " + acall_arg.substring(0,acall_arg_comma);
+			   }
+			   acall_arg_num++;
+        	   insert_acall_set(acall_arg_insert);
+               acall_arg = acall_arg.substring(acall_arg_comma+1);
+ 			   acall_arg_comma = acall_arg.indexOf(',');
+		 }
```

Comment:
Has the string been checked carefully before calling insert_acall_parms? If not, see next items.
1. Line 3234 checks for single quote. If found, it arbitrarily sets ending single quote on line 3235. What if the comma is within a quoted substring?
2. Line 3241 extracts substring following the found comma. What if the string ends with a comma? What if there were two commas in a row?

These may be okay. Just asking.